certbot/tests/ocsp_test.py

```diff
@@ -145,18 +146,22 @@ class OSCPTestCryptography(unittest.TestCase):
     """
 
     def setUp(self):
-        from certbot._internal import ocsp
+        from certbot import ocsp
         self.checker = ocsp.RevocationChecker()
         self.cert_path = test_util.vector_path('ocsp_certificate.pem')
         self.chain_path = test_util.vector_path('ocsp_issuer_certificate.pem')
         self.cert_obj = mock.MagicMock()
-        self.cert_obj.cert = self.cert_path
-        self.cert_obj.chain = self.chain_path
+        self.cert_obj.cert_path = self.cert_path
+        self.cert_obj.chain_path = self.chain_path
         now = pytz.UTC.fromutc(datetime.utcnow())
-        self.cert_obj.target_expiry = now + timedelta(hours=2)
-
-    @mock.patch('certbot._internal.ocsp._determine_ocsp_server')
-    @mock.patch('certbot._internal.ocsp._check_ocsp_cryptography')
+        self.mock_notAfter = mock.patch('certbot.ocsp.crypto_util.notAfter',
+                                        return_value=now + timedelta(hours=2))
+        self.mock_notAfter.start()
+        # Ensure the mock.patch is stopped even if test raises an exception
+        self.addCleanup(self.mock_notAfter.stop)
```

Comment:
Very cool! I had personally never seen this pattern before and we've been bitten by `stop` not being properly called if you just try and do it in `tearDown` as described at https://docs.python.org/3/library/unittest.mock.html#patch-methods-start-and-stop.

Nice work :+1: